version: '3.8'

services:
  db:
    image: postgres:15
    container_name: db
    volumes:
      - pg_data:/var/lib/postgresql/data/
    environment:
      - POSTGRES_USER=brunos1212
      - POSTGRES_PASSWORD=password
      - POSTGRES_DB=scrum
      - POSTGRES_HOST=db
    expose:
      - "5432"
    networks:
      - djangonetwork
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U brunos1212 -d scrum"]
      interval: 2s
      timeout: 1s
      retries: 5
  web:
    build: ../
    tty: true
    command: bash -c "python ./manage.py runserver 0.0.0.0:8000"
    container_name: django
    volumes:
      - ../:/usr/src/app/
    ports:
      - 8000:8000
    depends_on:
      db:
        condition: service_healthy
    networks:
      - djangonetwork
    env_file: "env-file"
volumes:
  pg_data:
networks:
      djangonetwork:
          driver: bridge